src/Mod/TechDraw/App/AppTechDrawPy.cpp

```diff
@@ -961,19 +961,17 @@ class Module : public Py::ExtensionModule<Module>
     {
         PyObject* pFace(nullptr);
         double scale = 1.0;
-        char* pPatName = "";
-        char* pPatFile = "";
+        const char* pPatName = {nullptr};
+        const char* pPatFile = {nullptr};
         TechDraw::DrawViewPart* source = nullptr;
         TopoDS_Face face;
 
-        if (!PyArg_ParseTuple(args.ptr(), "O|detet", &pFace, &scale, "utf-8", &pPatName, "utf-8", &pPatFile)) {
+        if (!PyArg_ParseTuple(args.ptr(), "O|dss", &pFace, &scale, &pPatName, &pPatFile)) {
             throw Py::TypeError("expected (face, [scale], [patName], [patFile])");

```

Comment:
Removing the automatic utf-8 encoding may lead to garbage path names. A better workaround might be to implement a helper function around PyMem_Free that accepts a `const char*`

Reply:
According to the docs, the 's' option provides a utf8 encoded c-string.  

's' for filenames is also used in makeWireString in Part.   If 's' is not suitable here, we should change it in Part also.

Reply:
I wonder whether "s" and "et" + "utf-8" do the same in Py3.x now. Under Linux I have made a test with a path name containing Japanese characters and this works with the old and new code:
```
import TechDraw

shape = Part.makeBox(1,1,1)
face = shape.Face1
TechDraw.makeGeomHatch(face, 1, "Diamond", "/tmp/ウィキペディアへようこそ/FCPAT.pat")
```

But this should be also tested under Windows to make sure the .pat file can be read-in.

Reply:
Using "s" seems to work the same as "et" on win.

On Win10 vm:
Directory & filename with odd characters works:
Z:/Patterns/FCPAT_ᐂᐙᑁᒔᓏᑍ.pat       (Diamond)
Z:/PatternsáéöĀ/FCPAT_ᐂᐙᑁᒔᓏᑍ.pat   (Diamond)

Pattern name with odd characters does not work (but does not work in v0.21 either)
Z:/Patterns/FCPAT.pat                (Diamondᐂᐙᑁᒔᓏᑍ) does not work (but does not work in v0.21)
Z:/Patterns/FCPAT.pat                (DiamondáéöĀ) does not work (but does not work in v0.21)

PR and v0.21 both give this error with odd pattern name:
15:08:59  Traceback (most recent call last):
15:08:59    File "C:\Users\wande\Documents\Target\FCDevl\Mod\Draft\draftguitools\gui_hatch.py", line 116, in onFileChanged
15:08:59      patterns = self.getPatterns(filename)
15:08:59    File "C:\Users\wande\Documents\Target\FCDevl\Mod\Draft\draftguitools\gui_hatch.py", line 127, in getPatterns
15:08:59      for line in patfile:
15:08:59    File "C:\Users\wande\Documents\Target\FCDevl\bin\lib\encodings\cp1252.py", line 23, in decode
15:08:59      return codecs.charmap_decode(input,self.errors,decoding_table)[0]
15:08:59  UnicodeDecodeError: 'charmap' codec can't decode byte 0x90 in position 34: character maps to <undefined>